framework/src/actions/CreateExecutionerAction.C

```diff
@@ -38,8 +40,14 @@ CreateExecutionerAction::act()
   _moose_object_pars.set<FEProblemBase *>("_fe_problem_base") = _problem.get();
   _moose_object_pars.set<FEProblem *>("_fe_problem") =
       (std::dynamic_pointer_cast<FEProblem>(_problem)).get();
+  _moose_object_pars.set<EigenProblem *>("_eigen_problem") =
+      (std::dynamic_pointer_cast<EigenProblem>(_problem)).get();
```

Comment:
I know this is a copy of the line above but it's either unsafe or incorrect. You are dynamic casting which suggests that you recognize that it might not always succeed, yet you assume it does by immediately dereferencing the pointer (which may be nullptr). I'd suggest either checking the cast, or changing to a static cast to indicate that you know that it will always succeed. You should change the other similar lines too.